app/assets/v2/js/cart-ethereum-polygon.js

```diff
@@ -0,0 +1,430 @@
+        .on('transactionHash', async(txHash) => {
+          console.log('Donation transaction hash: ', txHash);
+          _alert('Saving contributions. Please do not leave this page.', 'success', 2000);
+          await this.postToDatabase([txHash], bulkCheckoutAddressPolygon, userAddress); // Save contributions to database
+          await this.finalizeCheckout(); // Update UI and redirect
+        })
+        .on('error', (error, receipt) => {
+          // If the transaction was rejected by the network with a receipt, the second parameter will be the receipt.
+          this.handleError(error);
+        });
+    },
+
+    // Estimates the total gas cost of a polygon checkout and sends it to cart.js
+    async estimateGasCost() {
+      // The below heuristics are used instead of `estimateGas()` so we can send the donation
+      // transaction before the approval txs are confirmed, because if the approval txs
+      // are not confirmed then estimateGas will fail.
+      let networkId = appCart.$refs.cart.networkId;
+
+      if (networkId !== '80001' && networkId !== '137' && appCart.$refs.cart.chainId !== '1') {
+        return;
+      }
+
+      // If we have a cart where all donations are in Dai, we use a linear regression to
+      // estimate gas costs based on real checkout transaction data, and add a 50% margin
+      const donationCurrencies = this.donationInputs.map(donation => donation.token);
+      const daiAddress = this.getTokenByName('DAI')?.addr;
+      const isAllDai = donationCurrencies.every((addr) => addr === daiAddress);
+
+      if (isAllDai) {
+        if (donationCurrencies.length === 1) {
+          // Special case since we overestimate here otherwise
+          return 70000;
+        }
+        // TODO: find a suitable curve using
+        // https://github.com/mds1/Gitcoin-Checkout-Gas-Analysis
+        // return 27500 * donationCurrencies.length + 125000;
+        return 10000 * donationCurrencies.length + 70000;
+      }
+
+      // Otherwise, based on contract tests, we use the more conservative heuristic below to get
+      // a gas estimate. The estimates used here are based on testing the cost of a single
+      // donation (i.e. one item in the cart). Because gas prices go down with batched
+      // transactions, whereas this assumes they're constant, this gives us a conservative estimate
+      const gasLimit = this.donationInputs.reduce((accumulator, currentValue) => {
+        const tokenAddr = currentValue.token?.toLowerCase();
+
+        if (currentValue.token === MATIC_ADDRESS) {
+          return accumulator + 70000; // MATIC donation gas estimate
+
+        } else if (tokenAddr === '0x960b236A07cf122663c4303350609A66A7B288C0'.toLowerCase()) {
+          return accumulator + 170000; // ANT donation gas estimate
+        }
+
+        return accumulator + 70000; // generic token donation gas estimate
```

Comment:
do we wanna `70000` as a const and that there here

Reply:
the gas estimation work isn't accurate yet... more work on this in GITC-370 + GITC-372